torchvision/datasets/lfw.py

```diff
@@ -0,0 +1,260 @@
+            print('Files already downloaded and verified')
+            return
+        url = f"{self.download_url_prefix}{self.filename}"
+        download_and_extract_archive(url, self.root, filename=self.filename, md5=self.md5)
+        download_url(f"{self.download_url_prefix}{self.labels_file}", self.root)
+        if self.view == "people":
+            download_url(f"{self.download_url_prefix}{self.names}", self.root)
+
+    def _get_path(self, identity, no):
+        return os.path.join(self.images_dir, identity, f"{identity}_{int(no):04d}.jpg")
+
+    def extra_repr(self) -> str:
+        return f"Alignment: {self.image_set}\nSplit: {self.split}"
+
+    def __len__(self):
+        return len(self.data)
+
+
+class LFWPeople(_LFW):
+    """`LFW <http://vis-www.cs.umass.edu/lfw/>`_ Dataset.
+
+    Args:
+        root (string): Root directory of dataset where directory
+            ``lfw-py`` exists or will be saved to if download is set to True.
+        split (string, optional): The image split to use. Can be one of ``train`` (default), ``test``,
+            ``10fold``.
+        image_set (str, optional): Type of image funneling to use, ``original``, ``funneled`` or
+            ``deepfunneled``. Defaults to ``original``.
+        transform (callable, optional): A function/transform that  takes in an PIL image
+            and returns a transformed version. E.g, ``transforms.RandomRotation``
+        target_transform (callable, optional): A function/transform that takes in the
+            target and transforms it.
+        download (bool, optional): If true, downloads the dataset from the internet and
+            puts it in root directory. If dataset is already downloaded, it is not
+            downloaded again.
+
+    """
+
+    def __init__(
+        self,
+        root: str,
+        split: str = "train",
+        image_set: str = "funneled",
+        transform: Optional[Callable] = None,
+        target_transform: Optional[Callable] = None,
+        download: bool = False,
+    ):
+        super(LFWPeople, self).__init__(root, split, image_set, "people",
+                                        transform, target_transform, download)
+
+        self.class_to_idx = self._get_classes()
+        self.data, self.targets = self._get_people()
+
+    def _get_people(self):
+        data, targets = [], []
+        with open(os.path.join(self.root, self.labels_file), 'r') as f:
+            lines = f.readlines()
+            n_folds, s = (int(lines[0]), 1) if self.split == "10fold" else (1, 0)
+
+            for fold in range(n_folds):
+                n_lines = int(lines[s])
+                people = [line.strip().split("\t") for line in lines[s + 1: s + n_lines + 1]]
+                s += n_lines + 1
+                for i, (identity, num_imgs) in enumerate(people):
+                    for num in range(1, int(num_imgs) + 1):
+                        img = self._get_path(identity, num)
+                        if os.path.exists(img):
```

Comment:
Is there a possibility an image does not exist unless the data is corrupted?

Reply:
No. The downloaded files are completely fine but that just if incase someone alters/deletes/renames the images

Reply:
That would make matters worse: now the dataset would silently ignore some files and thus changing the dataset. Results that were trained on that might not be reproducible. So either we throw an error in case the file is not there, or we simply don't perform this check. In case the file is gone, the error will be expressive.

Reply:
Oh yes, you are right.
So I'll remove the check.
Or an alternative: 
create a list `self.missing = []`
```py
if os.path.exists(img):
    data.append(img)
    targets.append(self.class_to_idx[identity])
else:
   self.missing.append(img)
```
and in extra_repr
```py
def extra_repr(self) -> str:
    rpr =  super().extra_repr() + "\nClasses (identities): {}".format(len(self.class_to_idx))
    if len(self.missing) > 0:
        rpr + f"\nImages missing: {len(self.missing)}"
    return rpr
```
Let me know your thoughts, I'll update accordingly.

Reply:
I appreciate the thoroughness, but that again would hide the information in some sense. I'm guessing most users won't do `str(dataset)` and examine the result before they start training. Since we don't have any such checks in the other datasets, we can simply remove it.

Reply:
Okay Sir 👍